File: ts/project/.sdk/tm/java/utility/ResultHeaders.java
package JAVAPACKAGE.utility;

import java.util.LinkedHashMap;
import java.util.Map;

import JAVAPACKAGE.core.Context;
import JAVAPACKAGE.core.Helpers;
import JAVAPACKAGE.core.Response;
import JAVAPACKAGE.core.Result;

final class ResultHeaders {

  private ResultHeaders() {}

  static Result resultHeaders(Context ctx) {
    Response response = ctx.response;
    Result result = ctx.result;

    if (result != null) {
      if (response != null && response.headers != null) {
        Map<String, Object> hm = Helpers.toMapAny(response.headers);
        if (hm != null) {
          result.headers = hm;
        }
        else {
          result.headers = new LinkedHashMap<>();
        }
      }
      else {
        result.headers = new LinkedHashMap<>();
      }
    }

    return result;
  }
}

File: ts/project/.sdk/tm/rust/utility/result_headers.rs
use std::cell::RefCell;
use std::rc::Rc;

use crate::core::context::Context;
use crate::core::result::SdkResult;
use crate::utility::voxgigstruct::Value;

pub fn result_headers_util(ctx: &Rc<Context>) -> Option<Rc<RefCell<SdkResult>>> {
    let response = ctx.response.borrow().clone();
    let result = ctx.result.borrow().clone();

    if let Some(result) = &result {
        let headers = match &response {
            Some(r) => match &r.borrow().headers {
                Value::Map(m) => Value::Map(m.clone()),
                _ => Value::empty_map(),
            },
            None => Value::empty_map(),
        };
        result.borrow_mut().headers = headers;
    }

    result
}
